internal/app/connectconformance/results.go

```diff
@@ -382,3 +380,47 @@ func checkPayloads(expected, actual []*conformancev2.ConformancePayload) multiEr
 
 	return errs
 }
+
+func checkError(expected, actual *conformancev2.Error) multiErrors {
+	switch {
+	case expected == nil && actual == nil:
+		// nothing to do
+		return nil
+	case expected == nil && actual != nil:
+		return multiErrors{fmt.Errorf("received an unexpected error:\n%v", actual.String())}
+	case expected != nil && actual == nil:
+		return multiErrors{errors.New("expecting an error but received none")}
+	}
+
+	var errs multiErrors
+	if expected.Code != actual.Code {
+		errs = append(errs, fmt.Errorf("actual error code %d (%s) does not match expected code %d (%s)",
+			actual.Code, connect.Code(actual.Code).String(), expected.Code, connect.Code(expected.Code).String()))
+	}
+	if expected.Message != nil && expected.GetMessage() != actual.GetMessage() {
+		errs = append(errs, fmt.Errorf("actual error message %q does not match expected message %q",
+			actual.GetMessage(), expected.GetMessage()))
+	}
+	if len(expected.Details) != len(actual.Details) {
+		// TODO: Should this be more lenient? Are we okay with a Connect implementation adding extra
```

Comment:
IMO, this is OK. I think we just want to verify an implementation is able to specify error details. As long as they specify what we instruct them to, they're free to add anything on top of that (at least I think so).